remotetests/remotetest.js

```diff
@@ -885,6 +888,38 @@ describe('Caches', function() { //  it
       }
     });
   });
+
+  it('Create an Cache Resource with description and expiry',function(done){
+    var opts = baseOpts();
+    opts.cache = CACHE_RESOURCE_WITH_EXPIRY_NAME;
+    opts.description = CACHE_RESOURCE_WITH_EXPIRY_DESCRIPTION;
+    opts.cacheExpiryInSecs = CACHE_RESOURCE_WITH_EXPIRY_TIMEOUT;
+    apigeetool.createcache(opts,function(err,result) {
+      if (verbose) {
+        console.log('Create Cache result = %j', result);
+      }
+      if (err) {
+        done(err);
+      } else {
+        done()
+      }
+    });
+  });
```

Comment:
Thanks for adding tests, but we're not really testing that our options are working right.  It would be nice to see the test create a cache, retrieve it, verify the cache configuration and then delete it _(instead of having a test just to delete the created cache)_.  Here are the test scenarios I'd like to see:

1. Create cache with `cacheExpiryByDate`
2. Create cache with `cacheExpiryInSecs`
3. Create cache with both `cacheExpiryByDate` and `cacheExpiryInSecs`

Reply:
Currently the lib only supports create cache and delete cache. We would need to add retrieve cache command to verify the cache configuration.
Is it ok to update test with create and delete alone for these 3 scenarios?

Reply:
Ah, I didn't realize that.  Without the ability to get the cache to verify it's configuration, I'm not sure what else we can do other than do the create/delete instead of create/get/verify/delete.

Reply:
I do think one could easily make a _"raw"_ request to the Edge Management API to get the cache details.  By taking the [internal request code](https://github.com/apigee/apigeetool-node/blob/master/lib/commands/createcache.js#L66) and [making the request](https://apidocs.apigee.com/management/apis/get/organizations/%7Borg_name%7D/environments/%7Benv_name%7D/caches/%7Bcache_name%7D), it could be done.  This could be a good follow-up PR, the ability to get cache details/information.  _(I'm not suggesting it, but I can see this being useful.)_

Reply:
I have the requirement for get and should be working on it as next PR. For now should we update the test to check for create/delete cache? Pl confirm if this is fine.

Reply:
There are two paths:

1. The quick way:
    a. Create the 3 test scenarios mentioned above.  _(Each one is a separate test scenario)_
    b. Delete the standalone delete cache test _(Each test scenario should clean up itself)_
2. The right way:
    a. Everything in the quick way
    b. Write a function that gets a cache _(in `lib/commands/createcache.js`)_ so that you can verify the cache information
    c. Update the 3 test scenarios to be _create/get/verify/delete_ instead of _create/delete_

Ideally, I think the second option is best because the extra work required to do this would make a follow-up PR for `apigeetool getcache` really simple and it would prove that the new configuration options work as expected.  _(There is technically logic in your PR in that the presence of an option or options will impact the created cache.  We already know creating a cache works, so we need to verify the cache details.)_  And personally, the code would be very simple because you just need to use `defaults.defaultRequest(opts)` to create the request, make the request and verify.

Reply:
Test cases added using option 1 for below scenarios.

- Create cache with cacheExpiryByDate
- Create cache with cacheExpiryInSecs
- Create cache with both cacheExpiryByDate and cacheExpiryInSecs